### PR TITLE
Add Scala 3.3 Support

### PR DESCRIPTION
**Pull request by mdedetrich**
Aside from the expected mechanical work and some additional type ascriptions adding Scala 3 support here wasn't too difficult.

Resolves: https://github.com/apache/incubator-pekko-management/issues/79

### CODE REVIEW

File: discovery-aws-api/src/test/scala/org/apache/pekko/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscoveryTest.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ class FiltersParsingTest extends AnyFunSuite with Matchers {
   test("empty string does not break parsing") {
     val filters = ""
     val result: List[Filter] = parseFiltersString(filters)
-    result should be('empty)
+    result should be(Symbol("empty"))
```
Comment (pjfanning) [inline discovery-aws-api/src/test/scala/org/apache/pekko/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscoveryTest.scala:28]: I think `shouldBe empty` will work

Comment (mdedetrich) [reply discovery-aws-api/src/test/scala/org/apache/pekko/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscoveryTest.scala:28]: Isn't `shouldBe empty` for an empty collection/option? In this case we are checking for a string

File: management-cluster-bootstrap/src/main/scala/org/apache/pekko/management/cluster/bootstrap/contactpoint/HttpClusterBootstrapRoutes.scala [hunk 2]
```diff
@@ -78,6 +78,6 @@ object ClusterBootstrapRequests {
   import pekko.http.scaladsl.client.RequestBuilding._
 
   def bootstrapSeedNodes(baseUri: Uri): HttpRequest =
-    Get(baseUri + "/bootstrap/seed-nodes")
+    Get(baseUri.toString() + "/bootstrap/seed-nodes")
```
Comment (pjfanning) [inline management-cluster-bootstrap/src/main/scala/org/apache/pekko/management/cluster/bootstrap/contactpoint/HttpClusterBootstrapRoutes.scala:82]:
```suggestion
    Get(s"$baseUri/bootstrap/seed-nodes")
```

File: management-cluster-bootstrap/src/main/scala/org/apache/pekko/management/cluster/bootstrap/contactpoint/HttpClusterBootstrapRoutes.scala [hunk 3]
```diff
@@ -78,6 +78,6 @@ object ClusterBootstrapRequests {
...
```
Comment (pjfanning) [inline management-cluster-bootstrap/src/main/scala/org/apache/pekko/management/cluster/bootstrap/contactpoint/HttpClusterBootstrapRoutes.scala:81]:
```suggestion
    Get(s"$baseUri/bootstrap/seed-nodes")
```

Comment (mdedetrich) [reply management-cluster-bootstrap/src/main/scala/org/apache/pekko/management/cluster/bootstrap/contactpoint/HttpClusterBootstrapRoutes.scala]: Thanks fixed.

File: discovery-aws-api/src/test/scala/org/apache/pekko/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscoveryTest.scala (hunk 1 above)
Comment (mdedetrich) [reply discovery-aws-api/src/test/scala/org/apache/pekko/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscoveryTest.scala:28]: Ah I see what you mean, I thought that `Filter` was an alias for a Scala `Symbol` but in fact it is really checking if its empty. Seems like scalatest added these overloads for symbols which is unusual

Comment (mdedetrich) [reply discovery-aws-api/src/test/scala/org/apache/pekko/discovery/awsapi/ec2/Ec2TagBasedSimpleServiceDiscoveryTest.scala]: Fixed and pushed

File: lease-kubernetes/src/main/scala/org/apache/pekko/coordination/lease/kubernetes/LeaseActor.scala [hunk 4]
```diff
@@ -19,7 +19,6 @@ import org.apache.pekko
 import pekko.actor.Status.Failure
 import pekko.actor.{ ActorRef, DeadLetterSuppression, FSM, LoggingFSM, Props }
 import pekko.annotation.InternalApi
-import pekko.coordination.lease.kubernetes.LeaseActor._
```
Comment (mdedetrich) [inline lease-kubernetes/src/main/scala/org/apache/pekko/coordination/lease/kubernetes/LeaseActor.scala:22]: This was causing scala 3 to complain about a cyclic reference check hence why it was moved to be inside of `class LeaseActor`

**Comment by mdedetrich** [comment]
@nvollmar Due to how trivial it was to get Scala 3 to compile/test I went ahead and prototyped it to see if there was any potential problems.

While everything is compiling fine, there are tests that fail to pass just on Scala 3, i.e.

```
[error]         org.apache.pekko.management.cluster.bootstrap.SelfAwareJoinDeciderSpec
[error]         org.apache.pekko.management.cluster.bootstrap.SelfAwareJoinDeciderIPv6Spec
```

I am not that familiar with pekko-management so maybe this is something you can look into to?

**Comment by mdedetrich** [comment]
> @mdedetrich I've already solved that on my Scala 3 branch, I'll rebase upon yours

Awesome thanks. I have noticed that your banch seems to have more changes then what is necessary for Scala 3 support?

If so maybe as part of your rebasing only merge what is strictly necessary? Alternatively another way to do this is if you can point me to the fix I can add it to this branch, add co-authorship to you and then when my branch is merged then you can rebase with your additional improvements.

**Comment by nvollmar** [comment]
I fixed all warnings that popped up and enabled warn as errors.

**Comment by mdedetrich** [comment]
Thanks, I just read the merge commit you made recently in more detail and indeed its an improvement on my run. Running the tests and if its all good I will merge it into this PR and then we can do a squash + merge.

**Comment by mdedetrich** [comment]
@pjfanning @nvollmar PR is ready to review, note that the entire PR has already passed tests (see https://github.com/mdedetrich/incubator-pekko-management/pull/1)

**Comment by nvollmar** [comment]
@pjfanning We already agreed on this one, I applied further changes I did on top of it. I'll close mine.

File: lease-kubernetes-int-test/src/main/scala/org/apache/pekko/coordination/lease/kubernetes/LeaseTestSuite.scala [hunk 5]
```diff
@@ -47,7 +47,7 @@ object LeaseTestSuite {
       case Success(_) if !failed =>
         log.info("Test succeeded")
         CoordinatedShutdown(as).run(TestPassedReason)
-      case Success(_) if failed =>
+      case Success(_) =>
```
Comment (mdedetrich) [inline lease-kubernetes-int-test/src/main/scala/org/apache/pekko/coordination/lease/kubernetes/LeaseTestSuite.scala:50]: @nvollmar I just noticed this which seems to be a behaviour change, I will try and revert it and add a suppress warning incase Scala 3 complains.

**Review by pjfanning — CHANGES_REQUESTED** [review]
there is a compile issue in latest build

**Comment by mdedetrich** [comment]
>  Yes I know, still working on cleaning stuff up.

**Comment by mdedetrich** [comment]
@pjfanning All issues resolved.

**Review by pjfanning — APPROVED** [review]
lgtm

**Pull request merged**